File: frontend/src/types/research.ts
export interface SourceReference {
  name: string
  type: string
  as_of: string | null
}

export interface ResearchReportListItem {
  symbol: string
  name: string
  title: string
  industry: string | null
  generated_at: string | null
}

export interface ResearchReportListResponse {
  source: string
  items: ResearchReportListItem[]
  total: number
}

export interface ResearchReportDetail extends ResearchReportListItem {
  content_markdown: string
  sources: SourceReference[]
  disclaimer: string
}

File: backend/app/schemas/research.py
from datetime import datetime

from pydantic import BaseModel


class SourceReference(BaseModel):
    name: str
    type: str
    as_of: str | None = None


class ResearchReportListItem(BaseModel):
    symbol: str
    name: str
    title: str
    industry: str | None = None
    generated_at: datetime | None = None


class ResearchReportListResponse(BaseModel):
    source: str
    items: list[ResearchReportListItem]
    total: int


class ResearchReportDetail(ResearchReportListItem):
    content_markdown: str
    sources: list[SourceReference]
    disclaimer: str

